lib/batches/batch_spec.go

```diff
@@ -70,9 +71,55 @@ type WorkspaceConfiguration struct {
 }
 
 type OnQueryOrRepository struct {
-	RepositoriesMatchingQuery string `json:"repositoriesMatchingQuery,omitempty" yaml:"repositoriesMatchingQuery"`
-	Repository                string `json:"repository,omitempty" yaml:"repository"`
-	Branch                    string `json:"branch,omitempty" yaml:"branch"`
+	RepositoriesMatchingQuery string   `json:"repositoriesMatchingQuery,omitempty" yaml:"repositoriesMatchingQuery"`
+	Repository                string   `json:"repository,omitempty" yaml:"repository"`
+	Branches                  []string `json:"branches,omitempty" yaml:"branch"`
```

Comment:
I'm not 100% sure here, but shouldn't it be this?

```suggestion
	Branches                  []string `json:"branches,omitempty" yaml:"branches"`
```